js/src/Map.ts

```diff
@@ -100,119 +119,131 @@ export class LeafletMapModel extends widgets.DOMWidgetModel {
+      bottom: 0,
+      right: 0,
+      left: 9007199254740991,
+    };
+    [bounds, pixel_bounds] = Object.keys(views).reduce(
+      function (
+        bnds_pixbnds: [typeof bounds, typeof pixel_bounds],
+        key: string
+      ) {
+        const bnds = bnds_pixbnds[0];
+        const pixbnds = bnds_pixbnds[1];
+        //@ts-ignore
+        const obj = views[key].obj;
+        if (obj) {
+          const view_bounds = obj.getBounds();
+          bnds.north = Math.max(bnds.north, view_bounds.getNorth());
+          bnds.south = Math.min(bnds.south, view_bounds.getSouth());
+          bnds.east = Math.max(bnds.east, view_bounds.getEast());
+          bnds.west = Math.min(bnds.west, view_bounds.getWest());
+          const view_pixel_bounds = obj.getPixelBounds();
+          const top_left = view_pixel_bounds.getTopLeft();
+          const bottom_right = view_pixel_bounds.getBottomRight();
+          pixbnds.top = Math.min(pixbnds.top, top_left.y);
+          pixbnds.bottom = Math.max(pixbnds.bottom, bottom_right.y);
+          pixbnds.right = Math.max(pixbnds.right, bottom_right.x);
+          pixbnds.left = Math.min(pixbnds.left, top_left.x);
+        }
+        return [bnds, pixbnds];
+      },
+      [bounds, pixel_bounds]
+    );
+    this.set('north', bounds.north);
+    this.set('south', bounds.south);
+    this.set('east', bounds.east);
+    this.set('west', bounds.west);
+    this.set('top', pixel_bounds.top);
+    this.set('bottom', pixel_bounds.bottom);
+    this.set('right', pixel_bounds.right);
+    this.set('left', pixel_bounds.left);
   }
 }
 
 LeafletMapModel.serializers = {
-  ...widgets.DOMWidgetModel.serializers,
-  layers: { deserialize: widgets.unpack_models },
-  controls: { deserialize: widgets.unpack_models },
-  style: { deserialize: widgets.unpack_models },
-  default_style: { deserialize: widgets.unpack_models },
-  dragging_style: { deserialize: widgets.unpack_models },
+  ...DOMWidgetModel.serializers,
+  layers: { deserialize: unpack_models },
+  controls: { deserialize: unpack_models },
+  style: { deserialize: unpack_models },
+  default_style: { deserialize: unpack_models },
+  dragging_style: { deserialize: unpack_models },
 };
 
-export class LeafletMapView extends utils.LeafletDOMWidgetView {
-  initialize(options) {
+export class LeafletMapView extends LeafletDOMWidgetView {
+  obj: Map;
+  dirty: boolean;
+  map_container: HTMLDivElement;
+  map_child: HTMLDivElement;
+  layer_views: ViewList<LeafletLayerView>;
+  control_views: ViewList<LeafletControlView>;
+  model: LeafletMapModel;
+
+  initialize(options: WidgetView.IInitializeParameters<LeafletMapModel>) {
     super.initialize(options);
     // The dirty flag is used to prevent sub-pixel center changes
     // computed by leaflet to be applied to the model.
     this.dirty = false;
   }
 
   create_panes() {
-    const panes = this.model.get('panes');
+    // const panes = this.model.get('panes');
+    const panes = this.obj.getPanes();
```

Comment:
I'm not sure about the diff here? We should probably keep the `model.get('panes')`. According to the git blame, this "panes" feature was introduced in [this PR](https://github.com/jupyter-widgets/ipyleaflet/pull/999) when fixing [this issue](https://github.com/jupyter-widgets/ipyleaflet/issues/993). You may test this feature running the `examples/MapPanes.ipynb` example.